stdlib/public/core/AutoDiff.swift

```diff
@@ -539,7 +539,10 @@ extension _AnyDerivativeBox {
 internal func _derivativeTypeMismatch(
   _ x: Any.Type, _ y: Any.Type, file: StaticString = #file, line: UInt = #line
 ) -> Never {
-  fatalError("Derivative type mismatch: \(x) and \(y)", file: file, line: line)
+  fatalError("""
```

Comment:
I think this should be a `preconditionFailure` in all cases.

Reply:
Done.